allennlp/common/util.py

```diff
@@ -106,12 +106,18 @@ def group_by_count(iterable: List[Any], count: int, default_value: Any) -> List[
 A = TypeVar("A")
 
 
-def lazy_groups_of(iterator: Iterator[A], group_size: int) -> Iterator[List[A]]:
+def lazy_groups_of(iterable: Iterable[A], group_size: int) -> Iterable[List[A]]:
```

Comment:
The return type here shouldn't be `Iterable` given that we're aiming for laziness. I think you want to use `Generator`.

Reply:
(Or just stick with `Iterator`. The fact that it's a generator probably shouldn't matter.)

Reply:
A `Generator` is an `Iterator`. So to make the minimum number of promises to the caller, I'll change it back to `Iterator`.